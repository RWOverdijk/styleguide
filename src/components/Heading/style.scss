/** @define Heading */

@import "../../css/_global";

.Van-Heading {
  &--h1 {
    font-weight: 700;
    font-size: 36px;
    line-height: 44px;
  }

  &--h2 {
    font-weight: 700;
    font-size: 28px;
    line-height: 34px;
  }

  &--h3 {
    font-weight: 700;
    font-size: 24px;
    line-height: 29px;
  }

  &--h4 {
    font-weight: 600;
    font-size: 18px;
    line-height: 22px;
  }
}
