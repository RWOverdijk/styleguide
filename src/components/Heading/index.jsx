import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './style.css';

class Heading extends PureComponent {
  getClass(className) {
    let classes = ['Van-Heading'];

    if (this.props.className) {
      classes.push(this.props.className);
    }

    classes.push(className);

    return classes.join(' ');
  }

  getMarkup(props) {
    let markup = <h1 className={this.getClass('Van-Heading--h1')} {...props}>{this.props.children}</h1>;

    if (this.props.size === 'h2') {
      markup = <h2 className={this.getClass('Van-Heading--h2')} {...props}>{this.props.children}</h2>;
    } else if (this.props.size === 'h3') {
      markup = <h3 className={this.getClass('Van-Heading--h3')} {...props}>{this.props.children}</h3>;
    } else if (this.props.size === 'h4') {
      markup = <h4 className={this.getClass('Van-Heading--h4')} {...props}>{this.props.children}</h4>;
    }

    return markup;
  }

  render() {
    const {
      className,
      ...props
    } = this.props;

    return this.getMarkup(props);
  }
}

Heading.propTypes = {
  className: PropTypes.string,
  size: PropTypes.string
};

export default Heading;
