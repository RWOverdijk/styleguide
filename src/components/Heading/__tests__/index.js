import React from 'react';
import renderer from 'react-test-renderer';
import Heading from '../';

describe("<Heading>", () => {
  it("renders h1 Heading if does not pass a size", () => {
    const tree = renderer.create(
      <Heading>My Heading</Heading>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders h1 Heading size", () => {
    const tree = renderer.create(
      <Heading size="h1">My Heading</Heading>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders h2 Heading size", () => {
    const tree = renderer.create(
      <Heading size="h2">My Heading</Heading>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders h3 Heading size", () => {
    const tree = renderer.create(
      <Heading size="h3">My Heading</Heading>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders h4 Heading size", () => {
    const tree = renderer.create(
      <Heading size="h4">My Heading</Heading>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <Heading className="my-class">My Heading</Heading>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
