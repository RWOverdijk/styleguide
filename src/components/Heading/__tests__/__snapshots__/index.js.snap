// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading> renders h1 Heading if does not pass a size 1`] = `
<h1
  className="Van-Heading Van-Heading--h1"
  size="h1"
>
  My Heading
</h1>
`;

exports[`<Heading> renders h1 Heading size 1`] = `
<h1
  className="Van-Heading Van-Heading--h1"
  size="h1"
>
  My Heading
</h1>
`;

exports[`<Heading> renders h2 Heading size 1`] = `
<h2
  className="Van-Heading Van-Heading--h2"
  size="h2"
>
  My Heading
</h2>
`;

exports[`<Heading> renders h3 Heading size 1`] = `
<h3
  className="Van-Heading Van-Heading--h3"
  size="h3"
>
  My Heading
</h3>
`;

exports[`<Heading> renders h4 Heading size 1`] = `
<h4
  className="Van-Heading Van-Heading--h4"
  size="h4"
>
  My Heading
</h4>
`;

exports[`<Heading> renders with specific className 1`] = `
<h1
  className="Van-Heading my-class Van-Heading--h1"
  size="h1"
>
  My Heading
</h1>
`;
